src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -118,6 +130,13 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdFile.set(new File(project.buildDir, ".docker/${safeTaskPath}-imageId.txt"))
+
+        if(imageIdFile.get().asFile.exists()){
```

Comment:
This needs to go into the Closure of `upToDate` when because a user could decide to set their own `imageIdFile` value. Calling `get()` here would evaluate the field too early.

Reply:
If the file is not there, gradle will tell you the task is no up-to-date because the file is not there. When the image is not there, it will tell you that upToDateWhen failed. 
If I move everyting inside the upToDateWhen closure, there won't be a separate message for both cases. Is that ok with you?